internal/settings/settings.go

```diff
@@ -182,64 +155,83 @@ func (f *flags) ConfMapConverters() []confmap.Converter {
 	return confMapConverters
 }
 
-func (f *flags) ServiceArgs() []string {
-	return f.serviceArgs
+// ColCoreArgs returns list of arguments to be passed to the collector core service.
+func (s *Settings) ColCoreArgs() []string {
+	return s.colCoreArgs
 }
 
-func (f *flags) IsDryRun() bool {
-	return f.dryRun
+// IsDryRun returns whether --dry-run mode was requested
+func (s *Settings) IsDryRun() bool {
+	return s.dryRun
 }
 
-func newFlags(args []string) (*flags, error) {
-	flagSet := flag.FlagSet{}
-	// we don't want to be responsible for tracking all supported collector service
-	// flags, so allow any we don't use and defer parsing to the actual service
-	flagSet.ParseErrorsWhitelist.UnknownFlags = true
-
-	var cpArgs []string
-	cpArgs = append(cpArgs, args...)
+// parseArgs returns new Settings instance from command line arguments.
+func parseArgs(args []string) (*Settings, error) {
+	flagSet := flag.NewFlagSet("otelcol", flag.ContinueOnError)
```

Comment:
As commented the aim here is to not have to track or manage upstream options in our init logic unless we are using them directly. Why is this no longer desired?

Reply:
>The reason for that is that there are no plans in the collector core to add any new flags.

Do you have an issue for this?

>it potentially can be incompatible with our distro and introduce some broken flags to the --help output without us noticing.

Do you mind clarifying why this expectation is something to avoid and why it's better to not adopt features as upstream provides them?

Reply:
> Do you have an issue for this?

No

> Do you mind clarifying why this expectation is something to avoid and why it's better to not inherit* features as upstream provides them?

Because additional flags introduce some optional functionality. And that functionality potentially can be incompatible with our distro, which will lead to an additional broken flag in `--help` output without us noticing. If that flag, for whatever reason, is required, we will catch it during the upgrade of the core dependency

Reply:
If we still want to define all known flags (including core's) in our distro but pass everything that we don't know to the core just to potentially catch a new flag added to the core without documenting it, we can do that. But it will require manual argument parsing. I don't think it is worth it. I'd rather keep an eye on new flags added to the core (which is very unlikely to happen any time soon) and add them properly along with documentation in our distro, the same way as we do for components.

Reply:
I'm still under the impression we probably don't want to define any flags we aren't using and the current problems we are running into are longstanding implementation defects. Are we not already manually parsing args atm and in your changes? I can see an inverse `flagSetToArgs()` that removes our distro-only flags and values as a major improvement to the current `removeFlag()`.

Reply:
> I'm still under the impression we probably don't want to define any flags we aren't using and the current problems we are running into are longstanding implementation defects. 

Speaking about `--help` documentation, we either define flags it in our distro or use the core's definition. We cannot do both. Given that we have a lot of additional flags in our distro and core `--config` flag is different, I don't think we should use core. So I believe we need to define it in our distro. If we are doing so, we should do it for all the flags. Why would we skip supported flags that are implemented in a different place like `--version` and `--feature-gates`?

> Are we not already manually parsing args atm and in your changes? I can see an inverse flagSetToArgs() that removes our distro-only flags and values as a major improvement to the current removeFlag().

We are not, `flagSetToArgs` is a reverse operation. It just generated specific arguments from parsed flags. All the parsing is done by the `pflag` library

Reply:
> We are not, flagSetToArgs is a reverse operation. It just generated specific arguments from parsed flags. All the parsing is done by the pflag library

We are using it to determine filter the args to pass to the otelcol service, which I would call manual parsing regardless of the library doing it. My suggestion (unvetted) is to use what you're doing but inverting it to remove our distro specific args (maybe w/ a shlex-like dep?) to generate the collector service args.

Reply:
Can you elaborate on reasoning behind this suggestion? Why do we want to pass all unknown arguments?

Reply:
I think on the whole I'm under the impression we should thinly wrap the actual otelcol service/cobra command using as reasonably minimal flag logic as possible. Unfortunately this is kind of opposed to how we are currently doing things and and have done them in the past, for various reasons. These gh issues suggest to me we've previously been overzealous w/ flag parsing and not great supporting otelcol service args or expected startup logic: https://github.com/signalfx/splunk-otel-collector/issues/97, https://github.com/signalfx/splunk-otel-collector/issues/660, and for https://github.com/signalfx/splunk-otel-collector/issues/2213 your recent PR gives another example of how tracking upstream flags can become easily stale: https://github.com/signalfx/splunk-otel-collector/pull/2339.

Reply:
As another anecdote, I think the removal of the overwrite properties converter and the new "set" cobra function will require us to either drop our "set" flag option or port the converter to our distro before stripping the "set" options. I can see advantages with either approach (no loss in existing functionality but more maintenance requirements vs. potential breaking changes but more reliance on upstream functionality).

Reply:
We cannot tinily wrap  the actual otelcol service/cobra command until we at least support the same set of flags which is not the case for `--config` and make sure that out custom flags are not needed to be displayed. I'm all in with that approach but not sure we want to keep broken `--help` output until that.

As for `--set` I believe we should mark it as deprecated and ask users to use `--config yaml:...` instead. But for that, we need support of config schemas first.

Reply:
why would `--set` be deprecated if it's now in the core flags? https://github.com/open-telemetry/opentelemetry-collector/blob/main/otelcol/flags.go#L51

What are you referring to by not supporting "config schemas"? We do have config providers that are selected by the config schemes if that's what you mean https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/main.go#L76 and we should fix our incorrect config flag logic if it doesn't reflect what is supported by core.

Reply:
> why would --set be deprecated if it's now in the core flags? https://github.com/open-telemetry/opentelemetry-collector/blob/main/otelcol/flags.go#L51

To avoid having workarounds. It's still unclear if we will keep `--set` in the core. But we can leave it's support in out distro also, we just need to copy the code from the core for now.

> What are you referring to by not supporting "config schemas"? We do have config providers that are selected by the config schemes if that's what you mean https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/main.go#L76 and we should fix our incorrect config flag logic if it doesn't reflect what is supported by core.

Yes, I mean config schemas in `--config` flag. Currently, our distro shows the following help output for that flag:

```
      --config --config=file:/path/to/first --config=file:path/to/second   Locations to the config file(s), note that only a single location can be set per flag entry e.g. --config=file:/path/to/first --config=file:path/to/second. (default [])
```

But if I run ` splunk-otel-collector git:(main) ./bin/otelcol --config=file:./conf.yaml`, I get 
```
2022/12/14 11:16:11 main.go:47: invalid settings detected: unable to find the configuration file file:./conf.yaml
```

Do you want to fix `--config` flag sooner than merging this PR? Also, we will never be able to show any help for our own flag, do we want that?

Reply:
Yes, I think not treating scheme-prefixed config values as valid for their latest use in the resolver uris is a bug and should be addressed. We can access the otelcol command flags so would be able to provide their output in help when we're ready to advertise them (assuming there isn't some indirection/internalization planned in the future):

```diff
 func runInteractive(settings service.CollectorSettings) error {
        cmd := service.NewCommand(settings)
+       cmd.Flags().Bool("some-made-up-flag", true, "this is in the collector service help content.")
        if err := cmd.Execute(); err != nil {
                return fmt.Errorf("application run finished with error: %w", err)
        }
```

```
bin/otelcol -h
Usage:
  otelcol [flags]

Flags:
      --config --config=file:/path/to/first --config=file:path/to/second   Locations to the config file(s), note that only a single location can be set per flag entry e.g. --config=file:/path/to/first --config=file:path/to/second. (default [])
      --feature-gates Flag                                                 Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature.
  -h, --help                                                               help for otelcol
      --set func                                                           Set arbitrary component config property. The component has to be defined in the config file and the flag has a higher precedence. Array config properties are overridden and maps are joined. Example --set=processors.batch.timeout=2s
      --some-made-up-flag                                                  this is in the collector service help content. (default true)
  -v, --version                                                            version for otelcol
```

Reply:
We need the flags in advance, accessing `cmd.Flags()` will only add the `--help` output, so we need to duplicate all the definitions before that. It's a hacky but good workaround which we can add later on after we fix `--config` flags. We can still merge this one and make those changes as the following steps:
1. Fix out --config flag
2. Migrate to this proposed workaround
WDYT?

Reply:
I think it'd be best to have a planning issue where we plan to separate pre and in-service flags, auditing why we do what we do in general (some of which likely has an expired rationale), and investigate whether its necessary (or wise) to wrap any collector flag.

Would you be ok with this pr fixing the config scheme check, migrating the parsing logic to the struct ~constructor so we can use its exposed fields, and not adding any new flags?

Reply:
Not sure what to plan here. We have a few issues with parsing, I'm suggesting fixing them in 3 steps:
1. Merge this PR to fix the`--config` doc, add missing docs, and fix parsing arguments with "=".
2. Add support to schemas in `--config`
3. Update flags parsing to use `--help` output from core

I still disagree that we should pass all unknown flags, we can discuss that separately and apply another PR if needed.

Reply:
And want the passing all unknown flags to be a separate PR to assess how much complexity it adds

Reply:
Actually, if we don't pass unknown attributes in step 3, we can end up with some "unsupported" arguments that potentially can be added to docs in the core but removed on our side. So I can do that along with the PR for step 3, and we can discuss there if worth doing or not

Reply:
If we are fronting the collector service properly it will support its own args as it does for the dep version we use. Our distro shouldn't take responsibility of it imo.

Reply:
maybe it makes sense to fix the config parsing and other issues alone since we aren't in alignment on how to move forward otherwise.

Reply:
Fixing each issue alone requires adding more hacks on top of what we have. I tried to come up with one set of related changes that doesn't require doing that.